packages/augur-core/source/libraries/DeploymentVerifier.ts

```diff
@@ -0,0 +1,161 @@
+            if (expectedAddress !== registeredAddress) return `CONTRACT ${name} HAS INCORRECT ADDRESS REGISTERED`;
+        }
+        return SUCCESS;
+    }
+
+    // Verify all initializable contracts are initialized
+    async verifyInitializations(): Promise<string> {
+        for (const name of INITIALIZED_CONTRACTS) {
+            const contract = new Initializable(this.dependencies, this.addresses[name]);
+            console.log(`Verifying initialization of ${name} at ${contract.address}`);
+            const initialized = await contract.getInitialized_();
+            if (!initialized) return `CONTRACT ${name} was not initialized`;
+            const augurRef = await contract.augur_();
+            if (augurRef !== this.augur.address) return `CONTRACT ${name} HAD BAD AUGUR REF ${augurRef}`;
+            if (TRADING_CONTRACTS.includes(name)) {
+                const augurTradingRef = await contract.augurTrading_();
+                if (augurTradingRef !== this.augurTrading.address) return `CONTRACT ${name} HAD BAD AUGUR TRADING REF ${augurTradingRef}`;
+            }
+        }
+        return SUCCESS;
+    }
+
+    async verifyAugurTradingInitialization(): Promise<string> {
+        const augurTradingAugurAddress = await this.augurTrading.augur_();
+        if (augurTradingAugurAddress !== this.augur.address) return `AugurTrading has Augur at ${augurTradingAugurAddress}. Expected: ${this.augur.address}`;
+        return SUCCESS;
+    }
+}
+
+export class Initializable extends Contract<BigNumber> {
+    public constructor(dependencies: Dependencies<BigNumber>, address: string) {
+		super(dependencies, address)
+    }
+    
+    public getInitialized_ = async (options?: { sender?: string }): Promise<boolean> => {
+		options = options || {}
+		const abi: AbiFunction = {"constant":true,"inputs":[],"name":"getInitialized","outputs":[{"internalType":"bool","name":"","type":"bool"}],"payable":false,"stateMutability":"view","type":"function"}
+		const result = await this.localCall(abi, [], options.sender)
+		return <boolean>result[0]
+    }
+    
+    public augur_ = async (options?: { sender?: string }): Promise<string> => {
+		options = options || {}
+		const abi: AbiFunction = {"constant":true,"inputs":[],"name":"augur","outputs":[{"internalType":"contract IAugur","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"}
+		const result = await this.localCall(abi, [], options.sender)
+		return <string>result[0]
+    }
+    
+    public augurTrading_ = async (options?: { sender?: string }): Promise<string> => {
+		options = options || {}
+		const abi: AbiFunction = {"constant":true,"inputs":[],"name":"augurTrading","outputs":[{"internalType":"contract IAugurTrading","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"}
+		const result = await this.localCall(abi, [], options.sender)
+		return <string>result[0]
```

Comment:
I'm not sure you need to specify `<string>` here since it's implied by the function's return type. But if you do then we've standardized on `result[0] as string` instead.

Reply:
ah this is just copying the format used by the GenericContractinterfaces